site/components/MdxComponents/MdxComponents.tsx

```diff
@@ -63,27 +63,51 @@ export const components = {
   ),
   h3: ({ id, ...props }) => {
     return (
-      <span id={id} style={{ scrollMarginTop: 100 }}>
-        <Text
-          as="h3"
-          marginBottom="2"
-          marginTop="11"
-          variant="title2"
-          weight="semibold"
-          {...props}
-        />
-      </span>
+      <Box
+        id={id}
+        marginBottom="2"
+        marginTop="11"
+        scrollMarginTop={{
+          lg: 'scrollTop1',
+          md: 'scrollTop2',
+          sm: 'scrollTop3',
+          xs: 'scrollTop3',
+        }}
+      >
+        <Link href={`#${id}`}>
+          <Text
+            as="h3"
+            display="inline"
+            variant="title2"
+            weight="semibold"
+            {...props}
+          />
+        </Link>
+      </Box>
     );
   },
-  h4: props => (
-    <Text
-      as="h4"
+  h4: ({ id, ...props }) => (
+    <Box
+      id={id}
       marginBottom="2"
       marginTop="10"
-      variant="title3"
-      weight="semibold"
-      {...props}
-    />
+      scrollMarginTop={{
+        lg: 'scrollTop1',
+        md: 'scrollTop2',
+        sm: 'scrollTop3',
+        xs: 'scrollTop3',
```

Comment:
i like the idea here, but for the sake of simplicity, I think what you had before is good enough - just choose a hardcoded value that works well across mobile/desktop. you had 100px before I believe?

Reply:
Have tried to be mobile/desktop compatible in hard-code way, but the overlap with `Header` is not looking nice. 

If still feel not simplicity, I can revert the old way and set to 140px (The best top space for mobile/desktop I think).

Reply:
I tried setting it to `12` ('128px') and it worked fine on both mobile and desktop, is that overlapping for you?

Reply:
I tried to align with the text in the sidebar 🤣 

Let me setting it to `12` quikly, should i move the `scrollMarginTop` to `unresponsiveProperties`?

Reply:
i don't mind, it's up to you